source/shared/cpp/ObjectModel/CollectionTypeElement.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "pch.h"
+#include "BaseCardElement.h"
+
+namespace AdaptiveSharedNamespace
+{
+    class CollectionTypeElement : public BaseCardElement
+    {
+    public:
+        CollectionTypeElement(const CollectionTypeElement&) = default;
+        CollectionTypeElement(CollectionTypeElement&&) = default;
+        CollectionTypeElement& operator=(const CollectionTypeElement&) = default;
+        CollectionTypeElement& operator=(CollectionTypeElement&&) = default;
+        ~CollectionTypeElement() = default;
+        CollectionTypeElement(CardElementType type, ContainerStyle style = ContainerStyle::None,
+            VerticalContentAlignment alignment = VerticalContentAlignment::Top);
+
+        ContainerStyle GetStyle() const;
+        void SetStyle(const ContainerStyle value);
+
+        VerticalContentAlignment GetVerticalContentAlignment() const;
+        void SetVerticalContentAlignment(const VerticalContentAlignment value);
+
+        bool GetPadding() const;
+        void SetPadding(const bool value);
+
+        bool GetBleed() const;
+        void SetBleed(const bool value);
+        // tells if current element can bleed 
+        // when GetCanBleed() return false and GetBleed() returns true 
+        // the user must also check if HostConfig has
+        // padding for card, the root, if the padding is allowed,
+        // the user has to interpret the returned value correctly
```

Comment:
>the user has to interpret the returned value correctly [](start = 11, length = 54)

maybe expand on this comment a little -- how should this be correctly interpreted?

Reply:
sure, card itself can have padding, in that case, if an element has both padding and a bleed flag , it should bleed even though canBleed is false.

---
In reply to: [259076511](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259076511) [](ancestors = 259076511)